File: src/main/java/no/nav/veilarboppfolging/repository/entity/MaalEntity.java
package no.nav.veilarboppfolging.repository.entity;


import java.time.ZonedDateTime;

public class MaalEntity {
    private long id;
    private String aktorId;
    private String mal;
    private String endretAv;
    private ZonedDateTime dato;

    public String getEndretAvFormattert() {
        return erEndretAvBruker() ? "BRUKER" : "VEILEDER";
    }

    public boolean erEndretAvBruker() {
        return endretAv != null && endretAv.equals(aktorId);
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/repository/entity/MaalEntity.kt
package no.nav.veilarboppfolging.repository.entity

import java.time.ZonedDateTime


data class MaalEntity(
    val id: Long,
    val aktorId: String,
    val mal: String,
    val endretAv: String?,
    val dato: ZonedDateTime
) {
    val endretAvFormattert: String
        get() = if (erEndretAvBruker()) "BRUKER" else "VEILEDER"

    fun erEndretAvBruker(): Boolean {
        return endretAv != null && endretAv == aktorId
    }

    fun oppdaterMedNyId(nyId: Long): MaalEntity = this.copy(id = nyId)
}